### PR TITLE
Kinesis Video Update

### PR DESCRIPTION
**Pull request by sirknightj**
*Issue #, if available:*

*Description of changes:*
Replace old KVS ABIs and update the JNI to match

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/internal/producer/jni/NativeLibraryLoader.java [hunk 1]
```diff
@@ -21,12 +21,12 @@
 import com.amazonaws.kinesisvideo.common.preconditions.Preconditions;
 
 import androidx.annotation.NonNull;
-//import javax.annotation.concurrent.ThreadSafe;
+//import androidx.annotation.AnyThread;
```
Comment (unicornss) [inline aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/internal/producer/jni/NativeLibraryLoader.java:24]: let us remove if it is not required

File: aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/producer/StreamDescription.java [hunk 2]
```diff
@@ -1,30 +1,15 @@
...
- * for the specific language governing permissions and
- * limitations under the License.
- */
-
 package com.amazonaws.kinesisvideo.producer;
 
 /**
  * Class to represent the stream description object coming from a successful call to the DescribeStream service API.
+ *
+ * @author dduc
```
Comment (unicornss) [inline aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/producer/StreamDescription.java:6]: /nit to remove

**Review by unicornss — COMMENTED** [review]
In summary, this brings in  JNI changes for the latest CPP/C Producer SDK which has added Audio track and important changes to streaminfo and stability fixes.

File: aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/internal/producer/jni/NativeLibraryLoader.java (hunk 1 above)
Comment (sirknightj) [reply aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/internal/producer/jni/NativeLibraryLoader.java]: Done

File: aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/producer/StreamDescription.java (hunk 2 above)
Comment (sirknightj) [reply aws-android-sdk-kinesisvideo/src/main/java/com/amazonaws/kinesisvideo/producer/StreamDescription.java]: Done

**Review by unicornss — APPROVED** [review]
Porting JNI changes from latest Producer CPP SDK was verified with the aws android kvs demo sample.

**Review by disa6302 — APPROVED** [review]

**Review by poojamat — APPROVED** [review]

**Pull request merged**